src/scripts/condor_submit.sh

```diff
@@ -142,6 +142,12 @@ else
     fi
 fi
 
+# Replace all instances of '$HOME' in the environment with the actual path
+# to the user's home dir. We only do this for '$HOME' instead of all vars
+# since it is the most likely to be the same on the submit host and WN.
+[[ "x$condor_expand_home_env" == "xyes" ]] && \
+    submit_file_environment=$(sed 's|$HOME|'"$HOME"'|' <<< $submit_file_environment)
```

Comment:
How many suggestions can we pack into a two line change?

```suggestion
[[ $condor_expand_home_env = yes ]] &&
    submit_file_environment=${submit_file_environment//'$HOME'/$HOME}
```

- `[[ ... ]]` vars on the left hand side do not need to be quoted; and do not need the ugly `x$var` hack for uber- sh-compatibility with obscure implementations of `[ ... ]`
- quoting a literal string that consists entirely of letters (eg `xyes`) is never necessary
- for the `test` command, the equality test operator is actually `=`, though `==` is a synonym in bash.  If you prefer `==`, that's fine, but in order to avoid confusion when `test` / `[ ... ]` are used, i recommend just sticking to `=`
- `&&` at end-of-line will apply to the next line without a trailing `\`
- your `sed` expression will only replace a single occurrence of `$HOME` in `$submit_file_environment`, since there is no `g` option specified at the end of the `s|||` expression
- simple string replacements can be done entirely in the shell process

Reply:
I suppose we don't *have* to continue blahp stylistic idiosyncrasies :D

Reply:
Hah!

Well actually elsewhere in this file the uses of the bash `[[ ... ]]` actually looked pretty clean.  There are still some tasteless examples of `[ ... ]`, but if you are going to use the nicer `[[ ... ]]`, you might as well let it sparkle a bit.